app/fetch/fetch.go

```diff
@@ -0,0 +1,54 @@
+package fetch
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+)
+
+// exported so visible to main package
+type Response struct {
+	Count int     `json:"count"`
+	Rate  float32 `json:"rate"`
+}
+
+var (
+	treesAPIURL = "https://api.ecosia.org/v1/trees/count"
+	//TreeData is exported so it can be accessed from the fetch package
+	TreeData Response
+)
+
+// NewRequest creates a http client and makes the request, it returns a response and an error
+func NewRequest() (resp *http.Response, err error) {
+	client := &http.Client{}
+	req, err := http.NewRequest("GET", treesAPIURL, nil)
+	if err != nil {
+		return resp, err
+	}
+	resp, err = client.Do(req)
+	return resp, err
+}
+
+// Fetch returns response from Trees API and stores it in the response variable.
+func Fetch(makeRequest func() (*http.Response, error)) (int, error) {
+	resp, err := makeRequest()
+
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	if resp.StatusCode != 200 {
+		return resp.StatusCode, nil
+	}
+
+	respBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return http.StatusInternalServerError, err
+	}
+
+	err = json.Unmarshal(respBytes, &TreeData)
```

Comment:
Be careful, this is a race condition that will cause (if you compile your source with --race):
```
WARNING: DATA RACE
Write at 0x000000fb6e28 by goroutine 68:
  reflect.Value.SetFloat()
      /usr/local/go/src/reflect/value.go:1533 +0xed
  encoding/json.(*decodeState).literalStore()
      /usr/local/go/src/encoding/json/decode.go:1011 +0x337f
  encoding/json.(*decodeState).value()
      /usr/local/go/src/encoding/json/decode.go:395 +0x2f2
  encoding/json.(*decodeState).object()
      /usr/local/go/src/encoding/json/decode.go:763 +0x233c
  encoding/json.(*decodeState).value()
      /usr/local/go/src/encoding/json/decode.go:381 +0xb0
  encoding/json.(*decodeState).unmarshal()
      /usr/local/go/src/encoding/json/decode.go:179 +0x259
  encoding/json.Unmarshal()
      /usr/local/go/src/encoding/json/decode.go:106 +0x1e1
  github.com/ecosia/go-prometheus-workshop/app/fetch.Fetch()
      /src/fetch/fetch.go:58 +0x2ff
```